src/js/index.js

```diff
@@ -1,81 +1,137 @@
-let menuList = [];
-let category = 'espresso';
-
-// events
-const editMenu = ({target}) => {
-  if (!target.classList.contains('menu-edit-button')) return;
-  const name = window.prompt('메뉴명을 수정하세요');
-  if (!name) return;
-
-  const menuId = Number(target.parentElement.dataset.menuId);
-  menuList = menuList.map((menu, index) => index === menuId ? {...menu, name} : menu);
-  renderAboutMenus();
-};
-
-const removeMenu = ({target}) => {
-  if (!target.classList.contains('menu-remove-button')) return;
-  if (!window.confirm('정말 삭제하시겠습니까?')) return;
-
-  const menuId = Number(target.parentElement.dataset.menuId);
-  menuList = menuList.filter((_, index) => index !== menuId);
-  renderAboutMenus();
-};
-
-const addMenuList = (name) => {
-  if (!name) return;
-  menuList = [...menuList, {name, category}];
-  renderAboutMenus();
-  $input.value = '';
-};
-
-// addEventListeners
-$form.addEventListener('submit', (e) => {
-  e.preventDefault();
-  addMenuList($input.value);
-});
-
-const addEventListenersToMenuList = () => {
-  const $menu = document.getElementById('espresso-menu-list');
-  $menu.addEventListener('click', editMenu);
-  $menu.addEventListener('click', removeMenu);
+import { $, $id } from './utils/dom.js';
+import LocalStorage from './utils/localStorage.js'
+
+function App () {
+  const MenuListStorage = new LocalStorage('menuList');
+
+  // states
+  // [{name: string; category: string; isSoldOut: boolean;}]
+  let menuList = MenuListStorage.get() ?? [];
+  let category = 'espresso';
+
+  window.onload = () => {
+    initEventListenes();
+    renderAboutMenus();
+  };
+
+  // utils
+  const getMenuId = (target) => {
+    return Number(target.parentElement.dataset.menuId);
+  };
+
+  // events
+  const addMenu = () => {
+    let name = $id('menu-name').value?.trim();
+    if (!name) return;
+
+    menuList = [...menuList, {name, category}];
+    MenuListStorage.set(menuList);
+    renderAboutMenus();
+    $id('menu-name').value = '';
+  };
+
+  const editMenu = ({target}) => {
+    const name = window.prompt('메뉴명을 수정하세요').trim();
+    if (!name) return;
+
+    const menuId = getMenuId(target);
+    menuList = menuList.map((menu, index) => index === menuId ? {...menu, name} : menu);
+    MenuListStorage.set(menuList);
+    renderAboutMenus();
+  };
+
+  const removeMenu = ({target}) => {
+    if (!window.confirm('정말 삭제하시겠습니까?')) return;
+
+    const menuId = getMenuId(target);
+    menuList = menuList.filter((_, index) => index !== menuId);
+    MenuListStorage.set(menuList);
+    renderAboutMenus();
+  };
+
+  const toggleIsSoldOut = ({target}) => {
+    const menuId = getMenuId(target);
+    menuList = menuList.map((menu, index) => 
+      index === menuId ? {...menu, isSoldOut: !menu.isSoldOut} : menu,
+    );
+    MenuListStorage.set(menuList);
+    renderAboutMenus();
+  };
+
+  const setCategory = ({target}) => {
+    const selectedCategory = target.dataset.categoryName;
+    category = selectedCategory;
```

Comment:
```suggestion
    category = target.dataset.categoryName;
```

이렇게 선언하지 않은건 `selectedCategory` 라는 변수 이름을 강조하고 싶으셨던건지 궁금해요~!